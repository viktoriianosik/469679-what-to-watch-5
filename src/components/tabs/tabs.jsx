import React, {Fragment} from "react";
import PropTypes from "prop-types";
import MoviePropTypes from "../movie/movie-props";
import ReviewPropTypes from "../movie-reviews/movie-review-props";
import {TABS} from "../../const";
import MovieDetails from "../movie-details/movie-details";
import MovieOverview from "../movie-overview/movie-overview";
import MovieReviews from "../movie-reviews/movie-reviews";
import {connect} from "react-redux";

const Tabs = (props) => {
  const tabs = Object.values(TABS);
  const {movie, reviews, activeTab, onTabClick} = props;

  return (
    <Fragment>
      <nav className="movie-nav movie-card__nav">
        <ul className="movie-nav__list">
          {tabs.map((tab, i) => (
            <li key={`movie-nav__item-${i}`} className={`movie-nav__item ${tab === activeTab ? `movie-nav__item--active` : ``}`}>
              <a href="#" className="movie-nav__link" onClick={(evt) => {
                evt.preventDefault();
                onTabClick(tab, movie.id);
              }}>{tab}</a>
            </li>
          ))}
        </ul>
      </nav>
      {activeTab === TABS.OVERVIEW && <MovieOverview movie={movie}/>}
      {activeTab === TABS.DETAILS && <MovieDetails movie={movie}/>}
      {activeTab === TABS.REVIEWS && <MovieReviews reviews={reviews}/>}
    </Fragment>
  );
};

Tabs.propTypes = {
  movie: MoviePropTypes,
  reviews: PropTypes.arrayOf(ReviewPropTypes).isRequired,
  activeTab: PropTypes.string.isRequired,
  onTabClick: PropTypes.func.isRequired,
};

const mapStateToProps = ({DATA}) => ({
  reviews: DATA.reviews,
});

export default connect(mapStateToProps)(Tabs);
