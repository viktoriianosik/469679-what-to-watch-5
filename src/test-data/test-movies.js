export const testMovies = [
  {
    backgroundColor: `#795433`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    description: `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    director: `Quentin Tarantino`,
    genre: `Crime`,
    id: 1,
    isFavorite: false,
    name: `Pulp Fiction`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 1.5,
    released: 1994,
    runTime: 153,
    scoresCount: 1635992,
    starring: [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  },
  {
    backgroundColor: `#BDAD8F`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/No_Country_for_Old_Men.jpg`,
    description: `Violence and mayhem ensue after a hunter stumbles upon a drug deal gone wrong and more than two million dollars in cash near the Rio Grande.`,
    director: `Ethan Coen`,
    genre: `Crime`,
    id: 14,
    isFavorite: false,
    name: `No Country for Old Men`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/No_Country_for_Old_Men.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/no-country-for-old-men.jpg`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 4.1,
    released: 2007,
    runTime: 122,
    scoresCount: 764976,
    starring: [`Tommy Lee Jones`, `Javier Bardem`, `Josh Brolin`],
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  }
];
