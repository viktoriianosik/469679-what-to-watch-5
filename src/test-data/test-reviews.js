export const testReviews = [
  {
    comment: `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    date: `2020-10-09T13:38:44.769Z`,
    id: 1,
    rating: 3.3,
    user: {
      id: 19,
      name: `Christina`
    },
  },
  {
    comment: `I love this movie.`,
    date: `2020-13-09T08:38:44.769Z`,
    id: 1,
    rating: 5,
    user: {
      id: 9,
      name: `Maks`
    },
  },
  {
    comment: `Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    date: `2019-30-08T13:38:44.769Z`,
    id: 14,
    rating: 8.3,
    user: {
      id: 3,
      name: `Olga`
    },
  },
];
