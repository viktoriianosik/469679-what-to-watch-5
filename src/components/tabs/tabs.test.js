import React from "react";
import renderer from "react-test-renderer";
import {Tabs} from "./tabs";
import {testMovies} from "../../test-data/test-movies";
import {testReviews} from "../../test-data/test-reviews";

describe(`Render component Tabs correctly`, () => {
  it(`Render component Tab Overview`, () => {
    const tree = renderer.create(
        <Tabs
          movie={testMovies[0]}
          reviews={testReviews}
          activeTab={`Overview`}
          onTabClick={() => {}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render component Tab Reviews`, () => {
    const tree = renderer.create(
        <Tabs
          movie={testMovies[0]}
          reviews={testReviews}
          activeTab={`Reviews`}
          onTabClick={() => {}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
