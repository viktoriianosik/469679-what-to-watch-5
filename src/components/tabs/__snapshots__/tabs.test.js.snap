// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component Tabs correctly Render component Tab Overview 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      1.5
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Bad
        
        
        
        
      </span>
      <span
        className="movie-rating__count"
      >
        1635992
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Quentin Tarantino
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        John Travolta, Uma Thurman, Samuel L. Jackson
        
      </strong>
    </p>
  </div>,
]
`;

exports[`Render component Tabs correctly Render component Tab Reviews 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-card__reviews movie-card__row"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Christina
            </cite>
            <time
              className="review__date"
              dateTime="2020-10-09T13:38:44.769Z"
            >
              October 9, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          3.3
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Olga
            </cite>
            <time
              className="review__date"
              dateTime="2019-30-08T13:38:44.769Z"
            >
              Invalid Date
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8.3
        </div>
      </div>
    </div>
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            I love this movie.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Maks
            </cite>
            <time
              className="review__date"
              dateTime="2020-13-09T08:38:44.769Z"
            >
              Invalid Date
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          5
        </div>
      </div>
    </div>
  </div>,
]
`;
